Sources/NIOSSHServer/RemotePortForwarding.swift

```diff
@@ -45,29 +48,33 @@ final class RemotePortForwarder {
             value: 1
         )
         .childChannelInitializer { childChannel in
-            let (ours, theirs) = GlueHandler.matchedPair()
-
-            // Ok, ask for the remote channel to be created. This needs remote half closure turned on and to be
-            // set up for data I/O.
-            let promise = loop.makePromise(of: Channel.self)
-            self.inboundSSHHandler.createChannel(
-                promise,
-                channelType: .forwardedTCPIP(
-                    .init(
-                        listeningHost: host,
-                        listeningPort: childChannel.localAddress!.port!,
-                        originatorAddress: childChannel.remoteAddress!
+            // Great, now we add the glue handler to the newly-accepted channel, and then we don't allow this channel to go
+            // active until the SSH channel has. Both should go active at once.
+            childChannel.eventLoop.makeCompletedFuture {
+                let (ours, theirs) = GlueHandler.matchedPair()
+
+                // Ok, ask for the remote channel to be created. This needs remote half closure turned on and to be
+                // set up for data I/O.
+                let promise = loop.makePromise(of: Channel.self)
+                loopBoundHandler.value.createChannel(
+                    promise,
+                    channelType: .forwardedTCPIP(
+                        .init(
+                            listeningHost: host,
+                            listeningPort: childChannel.localAddress!.port!,
+                            originatorAddress: childChannel.remoteAddress!
+                        )
                     )
-                )
-            ) { sshChildChannel, _ in
-                sshChildChannel.pipeline.addHandlers([DataToBufferCodec(), theirs]).flatMap {
-                    sshChildChannel.setOption(ChannelOptions.allowRemoteHalfClosure, value: true)
+                ) { sshChildChannel, _ in
+                    sshChildChannel.eventLoop.makeCompletedFuture {
+                        try sshChildChannel.pipeline.syncOperations.addHandlers([DataToBufferCodec(), theirs])
+                        _ = sshChildChannel.setOption(ChannelOptions.allowRemoteHalfClosure, value: true)
+                    }
```

Comment:
This is returns a future so is async but we're ignoring it which means the this ssh child channel initializer is returning earlier than it should (i.e. before we know the option has been set).

You can do similar to what was done previously and `flatMap` the `setOption` call from the `makeCompletedFuture`:

```swift
sshChildChannel.eventLoop.makeCompletedFuture {
  // ...
}.flatMap { 
  sshChildChannel.setOption(ChannelOptions.allowRemoteHalfClosure, value: true)
}
```